### ISSUE TITLE
[rush] Regression with rush install --from/--to

### ISSUE THREAD

**Issue by wbern** [microsoft/rushstack#2720]
## Summary
rush install --from my-package --to my-package

should result in (shortened)

15:14 $ rush --debug install --to my-package | grep "\-\-filter"
Invoking package manager: [...]node_modules/pnpm/bin/pnpm.js install --store [...]/common/temp/pnpm-store --no-prefer-frozen-lockfile --strict-peer-dependencies --recursive --link-workspace-packages false --filter ...my-package...

but right now we are only getting: `--filter my-package...`.

>To select a package and its dependent packages (direct and non-direct), prefix the package name with an ellipsis: ...<package_name>. For instance, this will run the tests of foo and all packages dependent on it:

This implies that the `--from` tag is broken, which was originally introduced in PR https://github.com/microsoft/rushstack/pull/2435.

## Repro steps
1. Run `rush install --from my-package --to my-package`
2. Verify that there are no leading `...` on the --filter parameter.

  **Expected result:** 

  **Actual result:** 

## Standard questions

Please answer these questions to help us investigate your issue more quickly:

| Question | Answer |
| -------- | -------- |
| `@microsoft/rush` globally installed version? | 5.47.0 |
| `rushVersion` from rush.json? | 5.47.0 |
| `useWorkspaces` from rush.json? | Yes |
| Operating system? | Linux  |
| Would you consider contributing a PR? | Not right now |
| Node.js version (`node -v`)? | 14.15.0 LTS |

**Comment by octogonz** [comment]
@D4N14L found that the code was accidentally deleted in PR https://github.com/microsoft/rushstack/pull/2486 which was refactoring the project selection logic.

@dmichon-msft  @iclanton could you take a look at this?

**Comment by wbern** [comment]
I read the PR description, and while it looks like --from and --to is replaced by other parameter names, I'm confused what all the "Unsafe" refers to, and perhaps why --to and --from weren't kept since they mean the same as in rush build params. Happy to hear more about reasoning. 🙂 

>--only: Install only the declared dependencies of the specified project
--impacted-by: Install the dependencies of the specified project and all projects that depend on it. Same as what --from previously did. Unsafe.
--impacted-by-except: Install the dependencies of all projects that depend on the specified project. Unsafe.
--to-except: Install the dependencies of all projects that the specified project directly or indirectly depends on.

**Comment by octogonz** [comment]
@wbern Read this doc: https://rushjs.io/pages/developer/selecting_subsets/

**Comment by octogonz** [comment]
I'm guessing that the "Unsafe" parameters would be a bad idea for `rush install`.  However the big design change was to make `--from` be safe, so I don't see why it would be removed.

**Comment by wbern** [comment]
Ah, I see. Great docs. 🙂

**Comment by dmichon-msft** [comment]
Are there any projects that depend on `my-project` in your workspace? The updated implementation of `rush install --from my-project` is implemented by computing the set of all projects that depend on `my-project`, then generating the install filter for `--to x` for each of them, since `--from` was redesigned to always select all dependencies of included projects.

**Comment by wbern** [comment]
Based on what you mentioned I decided to do some more digging, and I found out that during our "rush build" step, some uncommitted files were created (type definitions). These files were triggering our `git diff` mechanism for changed rush projects, and causing the issue where deps weren't installed for a particular package, but that package was later expected to run tests (as some file had changed inside the repo).

I'm closing this. Thanks!
